x/concentrated-liquidity/swaps.go

```diff
@@ -382,6 +382,15 @@ func (k Keeper) computeOutAmtGivenIn(
 				return sdk.Coin{}, sdk.Coin{}, PoolUpdates{}, sdk.Dec{}, err
 			}
 			swapState.tick = newTick
+
+			// if swapState.liquidity.IsPositive() && amountIn.IsZero() && !swapState.amountSpecifiedRemaining.IsZero() {
+			// 	return sdk.Coin{}, sdk.Coin{}, PoolUpdates{}, sdk.Dec{}, types.UnableToProgressSwapStep{
+			// 		Tick:              nextInitializedTick,
+			// 		SqrtPriceCur:      swapState.sqrtPrice,
+			// 		SqrtPriceNext:     computedSqrtPrice,
+			// 		AmountInRemaining: swapState.amountSpecifiedRemaining,
+			// 	}
+			// }
```

Comment:
I tried this in a bunch of places but there are many instances where we have valid swap step iterations with `amountIn` being zero. This doesn't work as an approach